public/app/features/dashboard/state/DashboardModel.ts

```diff
@@ -814,7 +815,7 @@ export class DashboardModel {
       const panel = this.panels[index];
 
       // break when encountering another row
-      if (panel.type === 'row') {
+      if (panel.type === 'row' || _.findIndex(this.panels[rowIndex].panels, ['id', panel.id]) === -1) {
```

Comment:
this condition does not make any sense (won't work in the expanded state as panels is empty then) , sort of a chicken and egg problem you have here , how do you determine what goes into row.panels array? break when we find panel that does not belong to row.panels! circular logic (at least appears that way to me)

Reply:
Hi @torkelo, I prefer to review the PR with the latest commit in GH because I made all these changes six months ago and there is a lot of new changes on Grafana. I have very few free time but I'm going to try it.

However, if you see my first comments I suggested you some alternatives such as to add a new attribute in panels which shows the row of a panel belongs (if exists) and change collapse/expand methods according to this new attribute. If you're going to keep rows in Grafana, it's a cleaner solution.

Reply:
> I prefer to review the PR with the latest commit in GH  

? this is your latest commit , and we have not changed any of this in the meantime

Reply:
> However, if you see my first comments I suggested you some alternatives such as to add a new attribute in panels which shows the row of a panel belongs (if exists) and change collapse/expand methods according to this new attribute. If you're going to keep rows in Grafana, it's a cleaner solution. 

Yea, but that has huge UX challenges. But back to this PR this solution seems unworkable so closing this